### PR TITLE
Add Redactor skeleton for migrating redaction logic from event_parser

### PR DESCRIPTION
**Pull request by bduffany**
This PR is a NOP for now but lays the groundwork for us to be able to factor the redaction logic out of `StreamingEventParser.ParseEvent` (and into `StreamingRedactor.RedactMetadata`) without breaking anything.

---

**Version bump**: Patch

### CODE REVIEW

File: server/build_event_protocol/build_event_handler/build_event_handler.go [hunk 1]
```diff
@@ -72,11 +72,13 @@ func (b *BuildEventHandler) OpenChannel(ctx context.Context, iid string) interfa
 		chunkFileSizeBytes = defaultChunkFileSizeBytes
 	}
 	buildEventAccumulator := accumulator.NewBEValues(iid)
```
Comment (vadimberezniker) [inline server/build_event_protocol/build_event_handler/build_event_handler.go:74]: You can inline this.

File: server/build_event_protocol/build_event_handler/build_event_handler.go [hunk 2]
```diff
@@ -72,11 +72,13 @@ func (b *BuildEventHandler) OpenChannel(ctx context.Context, iid string) interfa
...
+	redactor := redact.NewStreamingRedactor(b.env)
```
Comment (vadimberezniker) [inline server/build_event_protocol/build_event_handler/build_event_handler.go:75]: Will the redactor contain any state? If not, should we instantiate a single instance in NewBuildEventHandler?

Comment (bduffany) [reply server/build_event_protocol/build_event_handler/build_event_handler.go]: Yes, on each build event stream, the BuildStartedEvent contains a piece of BuildMetadata called ALLOW_ENV and we need to store that value in the redactor so it knows which pieces of the env need to be redacted (the env comes in a subsequent event)

File: server/build_event_protocol/build_event_handler/build_event_handler.go (hunk 1 above)
Comment (bduffany) [reply server/build_event_protocol/build_event_handler/build_event_handler.go]: Done

**Review by vadimberezniker — APPROVED** [review]
LGTM

**Review by tylerwilliams — APPROVED** [review]
LGTM

**Pull request merged**